pkg/filters/filters.go

```diff
@@ -0,0 +1,168 @@
+package filters
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	InReg         = regexp.MustCompile(` in `)
+	ContainsReg   = regexp.MustCompile(` contains `)
+	InRegValue    = regexp.MustCompile(`(?s)\((.*)\)`)
+	termOperators = []string{NotEquals, Equals, GreaterThanEquals, GreaterThan, LessThanEquals, LessThan, Contains, In}
+)
+
+// SplitTerms split the filter string and returns the map of strings
+func SplitTerms(filter string) []string {
+	if filter != "" {
+		return strings.Split(filter, ",")
+	}
+	return []string{}
+}
+
+// Transform transform the field selector term from string to flyteadmin field selector syntax
+func Transform(filters []string) (string, error) {
+	adminFilter := ""
+	for _, f := range filters {
+		if lhs, op, rhs, ok := parse(f); ok {
+			unescapedRHS, err := UnescapeValue(rhs)
+			if err != nil {
+				return "", err
+			}
+			if ok := validate(lhs, rhs); ok {
+				transformFilter := transform(lhs, op, unescapedRHS)
+				if len(adminFilter) > 0 {
+					adminFilter = fmt.Sprintf("%v+%v", adminFilter, transformFilter)
+				} else {
+					adminFilter = fmt.Sprintf("%v", transformFilter)
+				}
+			} else {
+				// TODO(Yuvraj): Add filters docs in error
+				return "", fmt.Errorf("Please add a valid field selector")
+			}
+		}
+	}
+	return adminFilter, nil
+}
+
+// validate validate the field selector operation
+func validate(lhs, rhs string) bool {
+	// TODO Add Validation check with regular expression
+	if len(lhs) > 0 && len(rhs) > 0 {
+		return true
+	}
+	return false
+}
+
+// InvalidEscapeSequence indicates an error occurred unescaping a field selector
+type InvalidEscapeSequence struct {
+	sequence string
+}
+
+func (i InvalidEscapeSequence) Error() string {
+	return fmt.Sprintf("invalid field selector: invalid escape sequence: %s", i.sequence)
+}
+
+// UnescapeValue unescapes a fieldSelector value and returns the original literal value.
+// May return the original string if it contains no escaped or special characters.
+func UnescapeValue(s string) (string, error) {
```

Comment:
Add the missing coverage for the cases in this function. Lets try to 100 % coverage for this file as it become a core filter class used by all the other commands